### ISSUE TITLE
BrowserComponent.setUrl with local file (jar:...) does not work on iOS

### ISSUE THREAD

**Issue by javieranton-zz** [codenameone/CodenameOne#2577]
No matter what build hints are used (allow local, allow all urls) it doesn't work

Tested on iOS 11

It does work on android and when debugging on computer (Windows)

**Comment by javieranton-zz** [comment]
Add a sample html file to project file.html with following content

`<!DOCTYPE html>
`<html>
`<body>
`<div>IF YOU SEE THIS THERE IS NO ERROR</div>

</body>
</html>

Add this to a control's constructor

@Override
	public Container create() {
BrowserComponent browser = new BrowserComponent();
browser.setURL("jar:///file.html");
RelaGraph.getInstance().relaGraphContainer = BorderLayout.center(browser);
}

Now if you see the html text in the control there is no bug. If however you do not see it there is a bug (iOS)

**Comment by javieranton-zz** [comment]
Hi

Are there any plans to fix this?

**Comment by javieranton-zz** [comment]
Hi,

I did as suggested by Shai and installed native logs

The logs after the supposed internal iOS failure to load local HTML page are as follows:

2018-11-22 03:57:18.184 FamiliaFernandez[264:7924] [EDT] 0:0:0,3 - Codename One revisions: 3fe43b381af2e2c0c5ef915d713f16a112d9d5af
2018-11-22 03:57:18.206 FamiliaFernandez[264:7924] [EDT] 0:0:0,24 - Native Logs Reader initialized correctly


Therefore native logs don't show what the error is. What else can I do to fix this for you?

**Comment by codenameone** [comment]
Try adding the build hint `ios.plistInject=<key>NSAppTransportSecurity</key><dict><key>NSAllowsArbitraryLoads</key><true/></dict>` to see if it works around the issue

**Comment by codenameone** [comment]
Try using `jar://file.html` also make sure the file is valid for iOS. This one seems like it's missing viewport which is required for iOS rendering.

**Comment by javieranton-zz** [comment]
Hi,

I tried:
1. Adding Viewport tag in the following way (in header):
`<meta name="viewport" content="width=device-width, initial-scale=1">`
2. Using `jar://file.html` 
3. Allowed all combinations of above items with the 2 different plisInject build hints (allowslocal and allowsarbitrary)

And the browser still could not be seen. Native logs were the same as the ones copied above (no errors)
 Am I the only one having this issue?

**Comment by javieranton-zz** [comment]
Hi, are there any plans to at least confirm whether this occurs on your environment too?

**Comment by codenameone** [comment]
Sorry, it got buried under my tasks. This isn't something we were able to reproduce. Local HTML files work for us. We rely on that functionality for the html package feature which is something we use heavily.

**Comment by codenameone** [comment]
No. We usually use HTML hierarchies. The web demo of the kitchen sink used to do that if I remember correctly.

**Comment by javieranton-zz** [comment]
Hi again. Still facing this issue on iOS

Simplest code I've come up with that will show a blank screen

               `Form previous = getCurrentForm();
		Form hierForm = new Form("Settings", new BorderLayout());
		Container boxCenter = new Container(BoxLayout.y());
		
		BrowserComponent browser = new BrowserComponent();
		
		browser.setURL("https://www.google.com");
		
		boxCenter.add(browser);
		hierForm.add(BorderLayout.CENTER,boxCenter);
		
		hierForm.getToolbar().setBackCommand(" ", ee -> {
			previous.showBack();
		});
		hierForm.show();`

My iOS build hints:
ios.application_exists: false
ios.debug.teamId: <my team id>
ios.dsym: false
ios.includePush: true
ios.interface_orientation: UIInterfaceOrientationPortrait:UIInterfaceOrientationPortraitUpsideDown:UIInterfaceOrientationLandscapeLeft:UIInterfaceOrientationLandscapeRight
ios_NSCameraUsage: Some functionality of the application requires your camera
ios_NSPhotoLibraryUsage: Some functionality of the application requires access to your photo library
ios_plistInject: <key>NSAppTransportSecurity</key><dict><key>NSAllowsLocalNetworking</key><true/></dict>
ios.prerendered_icon: false
ios.project_type: ios
ios.statusbar_hidden: false
ios.testFlight: false

**Comment by javieranton-zz** [comment]
I placed this simple code at the app's Init and this time it worked. Something in/after init must be blocking this. I will investigate

**Comment by javieranton-zz** [comment]
It has been a long time since I first opened this issue. I managed to understand how to make this work on iOS a while back. Basically, a html document can't be opened with `browser.setURL("jar:...` on iOS. Instead, a stream needs to be opened to a file placed in `src` directory (`Display.getInstance().getResourceAsStream(this.getClass(), "/Filename.html");` and then the string read from it needs to be injected via `browser.setPage(string, null);`. Only other way is to point the URL to a file on a https source. This has been the result of my very thorough testing of this
